src/XMakeCommandLine/XMake.cs

```diff
@@ -226,6 +226,21 @@ private static void HandleConfigurationException(Exception ex)
         }
 
         /// <summary>
+        /// Insert the command executable path as the first element of the args array.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static string[] ConstructArrayArg(string[] args)
```

Comment:
@AndyGerlicher Always using args[] would be nice but I thought it would be a compat risk since the args parsing in MSBuild might be different from the parsing .NET does to create the args to pass to the main method.